phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/SettingsView.java

```diff
@@ -105,6 +111,38 @@ private void initializeSettingsItems() {
                         tr("About")));
     }
 
+    private RepositoriesPanel createRepositoriesPanel() {
+        ObservableList<RepositoryLocation<? extends Repository>> repositoryLocations = FXCollections
+                .observableArrayList(settingsManager.loadRepositoryLocations());
+
+        final RepositoriesPanel repositoriesPanel = new RepositoriesPanel(repositoryLocationLoader,
+                repositoryLocations);
+
+        // set the initial values
+        repositoriesPanel.setOnRepositoryRefresh(repositoryManager::triggerRepositoryChange);
+
+        // react on changes
+        repositoriesPanel.getRepositoryLocations()
+                .addListener((ListChangeListener.Change<? extends RepositoryLocation<? extends Repository>> change) -> {
```

Comment:
This is not the cleanest way to do this.

A more solid and clean solution would be to use a `MappedList` to map the `RepositoryLocation` objects to `Repository` objects. The issue with this approach is, that the `RepositoryManager` would need to contain an `ObservableList` internally. With our current structure this is a bad idea, because the `RepositoryManager` is also used from the `cli` tool, which shouldn't depend on JavaFX.

A possible solution would be to implement two `RepositoryManager` classes: one for the javafx application and a second one for the cli application.

@qparis @plata what do you think?

Reply:
Don't think it's a good idea to have two `RepositoryManager` classes.